pkg/validators/am0006_dms_snitchnamepostfix.go

```diff
@@ -0,0 +1,30 @@
+package validators
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+func init() {
+	Registry.Add(AM0006)
+}
+
+var AM0006 = utils.Validator{
+	Code:        "AM0006",
+	Name:        "dms_snitchnamepostfix",
+	Description: "Ensure `deadmanssnitch.snitchNamePostFix` doesn't begin with 'hive-'",
+	Runner:      ValidateDmsSnitchNamePostFix,
+}
+
+func ValidateDmsSnitchNamePostFix(metabundle utils.MetaBundle) (bool, string, error) {
+	dmsConf := metabundle.AddonMeta.DeadmansSnitch
+	if dmsConf == nil || dmsConf.SnitchNamePostFix == nil {
```

Comment:
We are accepting no `dmsConf` or empty `SnitchNamePostFix`. Is this intended? I have no context on what we want from DMS, and want to enforce for our tenants. 

cc @apahim 

Are there other things we want to validate in the `DeadmansSnitch` config?

Reply:
Yep. All the fields under the `deadmanssnitch` section are templatised as a default manner. Although, we still give the option to the tenants to override the child fields like `snitchNamePostFix` in which case the above check makes sense.

> Are there other things we want to validate in the DeadmansSnitch config?
For now, none. I can think of one thing and that is that if `deadmanssnitch` field is there, then`deadmanssnitch.tags` must be there as well.

But the field level checks which will be added later should suffice for that.


https://github.com/mt-sre/addon-metadata-operator/blob/master/pkg/mtsre/v1/addons.go#L92-L93

Reply:
Interesting, actually I opened #42  to remove the field-level checks, as it seemed they were not required anymore. I felt having both `validate:"required"` tags and `// +kubebuilder:validation:Required` annotations was a duplication of the same feature.

The k8s API server will enforce all those kubebuilder annotations upon CR creation. 

WDYT?